cli/lsp/documents.rs

```diff
@@ -1257,6 +1257,7 @@ impl deno_graph::ModuleParser for LspModuleParser {
   }
 }
 
+#[allow(clippy::box_default)]
```

Comment:
A good rule of thumb is not to add supressions, but rather change the code as clippy suggests - in 90% cases clippy suggests a better solution than is currently applied, resulting in better code overall. What's the particular problem here?

Reply:
> A good rule of thumb is not to add supressions, but rather change the code as clippy suggests - in 90% cases clippy suggests a better solution than is currently applied, resulting in better code overall.

I agree with you.

> What's the particular problem here?

```rs
// deno_graph
impl CapturingModuleAnalyzer {
  pub fn new(
    parser: Option<Box<dyn ModuleParser>>,
    store: Option<Box<dyn ParsedSourceStore>>,
  ) -> Self {
    Self {
      parser: parser
        .unwrap_or_else(|| Box::new(DefaultModuleParser::default())),
      store: store
        .unwrap_or_else(|| Box::new(DefaultParsedSourceStore::default())),
    }
  }
}

// deno/lsp/document.rs
deno_graph::CapturingModuleAnalyzer::new(
    Some(Box::new(LspModuleParser::default())),
    None,
);
```

I think `LspModuleParser::default()` is different `DefaultModuleParser::default()`.

TBH, I did not know how to resolved this `clippy error`, so I just add `#[allow(clippy::box_default)]` code...

Could you please give me some help? I'm willing to try.

Reply:
It appears you should replace it like so:
```
Self {
      parser: parser
        .unwrap_or_else(|| Box::<DefaultModuleParser>::default()),
      store: store
        .unwrap_or_else(|| Box::<DefaultParsedSourceStore>::default()),
    }
```

Reply:
> It appears you should replace it like so:

Thank you for your help, I change the code to test in this repo.
I was test in my repo, but it always failed in `fastci` test case... 
Hopefully it won't take up too many CI resources.